### PR TITLE
Gerrit processes instances in parallel

### PR DESCRIPTION
**Pull request by chaodaiG**
This should improve the processing efficiency of gerrit

Pair-programmed with @mpherman2 
/cc @cjwagner

### CODE REVIEW

File: prow/gerrit/adapter/adapter.go [hunk 1]
```diff
@@ -222,44 +223,76 @@ func createCache(cloneURI *url.URL, cookieFilePath string, cacheSize int, config
...
-				c.repoCacheMap[cloneURI.String()] = cache
-			}
+				repoCacheMapMux.Unlock()
 
-			result := client.ResultSuccess
-			if err := c.processChange(log, instance, change, cloneURI, cache); err != nil {
-				result = client.ResultError
-				log.WithError(err).Info("Failed to process change")
+				result := client.ResultSuccess
+				if err := c.processChange(log, instance, change, cloneURI, cache); err != nil {
```
Comment (cjwagner) [inline prow/gerrit/adapter/adapter.go:262]: `cache` can be used concurrently safely, right? I think so, but just want to confirm.

File: prow/gerrit/adapter/adapter.go [hunk 2]
```diff
@@ -222,44 +223,76 @@ func createCache(cloneURI *url.URL, cookieFilePath string, cacheSize int, config
...
+					result = client.ResultError
+					log.WithError(err).Info("Failed to process change")
+				}
+				prometheusMux.Lock()
```
Comment (cjwagner) [inline prow/gerrit/adapter/adapter.go:266]: `prometheusMux` is not needed, updating prometheus metrics is threadsafe.

File: prow/gerrit/adapter/adapter.go [hunk 3]
```diff
@@ -222,44 +223,76 @@ func createCache(cloneURI *url.URL, cookieFilePath string, cacheSize int, config
...
+	go func() {
+		for err := range errChan {
+			errs = append(errs, err)
 		}
-		log.Infof("Processed %d changes", len(changes))
+	}()
+	wg.Wait()
+	close(errChan)
+
+	if len(errs) > 0 {
```
Comment (cjwagner) [inline prow/gerrit/adapter/adapter.go:292]: nit: You don't need to do this check. `utilerrors.NewAggregate(errs)` is smart enough to both ignore `nil` errors and return nil if there are no errors.

**Review by cjwagner — APPROVED** [review]
LGTM with some nits and a confirmation.
/hold

File: prow/gerrit/adapter/adapter.go (hunk 2 above)
Comment (chaodaiG) [reply prow/gerrit/adapter/adapter.go]: good to know, removed

File: prow/gerrit/adapter/adapter.go (hunk 3 above)
Comment (cjwagner) [reply prow/gerrit/adapter/adapter.go:290]: Oops yeah I actually saw that after commenting this then forgot to delete this comment. :man_facepalming:

**Review by cjwagner — APPROVED** [review]

**Pull request merged**